File: codigo/erlang/aula1.erl
-module(aula1).
-export([f/1, raiz_positiva/3, soma_N/1]).

f(X) ->
    X * 20.

% equacoes de segundo grau
delta(A, B, C) ->
    (B * B) - (4 * A * C).

raiz_positiva(A, B, C) ->
    (-B + math:sqrt(delta(A, B, C))) / (2 * A).

% soma dos N primeiros numeros
soma_N(N) when N =:= 1 -> 1;
soma_N(N) ->
    N + soma_N(N - 1).

File: codigo/elixir/aula1.ex
defmodule Aula1 do
  def f(x), do: x * 20

  def delta(a, b, c) do
    (b * b) - (4 * a * c)
  end

  def raiz_positiva(a, b, c) do
    (-b + :math.sqrt(delta(a, b, c))) / (2 * a)
  end

  def soma_N(1), do: 1
  def soma_N(n) do
    n + soma_N(n - 1)
  end
end
